README.md

```diff
@@ -1,53 +1,60 @@
-    ScaleWindow   - an allowed period for the resource to be inactive before calling the `Scale` function
-    MetricName    - name of the metric to monitor for the resource
-    Threshold     - A threshold for a metric to concider the resource being inactive
-    ScaleInterval - an interval to call scale function which checks if a resource has passed the scale window, if not the scale window is being considered the longest duration of time where the metric for the resource is under the threshold
-```
-Scaler should change the resource service to point to DLX service and not to the scaled down pod (to be changed by `ResourceScaler` or some other external entity)
-
-### DLX
-A service that listens on http requests for a given namespace, when a request is received, it's being saved in-memory until `Scale` function returns, this is an indication that the resource is ready and any changes that were made by autoscaler were reverted (i.e. changing where service points to), that request is then being proxied to the new instance and the response from it returns to the user.
+# Scale to Zero
 
+Infrastructure to scale any resource to and from zero
+
+## Prerequisites
+
+**Custom metrics API implementation:**
+
+The Autoscaler takes decisions based on data queried from Kubernetes 
+[custom metrics API.](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/instrumentation/custom-metrics-api.md)
+There are several possible tools that implement it, we internally use
+[Prometheus](https://prometheus.io/) with the [Prometheus-Adapter](https://github.com/DirectXMan12/k8s-prometheus-adapter)
+but you can use which ever you want... you can find [here](https://github.com/kubernetes/metrics/blob/master/IMPLEMENTATIONS.md#custom-metrics-api) 
+recommended implementations
+
+## Design:
+![Design scheme](https://www.lucidchart.com/publicSegments/view/cc8927a6-537f-4fe6-95e1-731503bc7996/image.jpeg)
+
+**The autoscaler** - responsible to periodically check whether resources should be scale to zero, it does that by 
+querying the custom metrics API. When it decide a resource should be scaled to zero it tells the resource scaler to 
+scale the resource to zero.
+The resource scaler will route the traffic to the DLX, which in terms of K8s is done by changing the svc selector, 
+after that, it will scale the resource to zero.
+
+**The DLX** - responsible for receiving and buffering requests of scaled to zero resources, when it gets a message it 
+creates a buffer for the messages, and tells the resource scaler to scale the service back from zero.
+The resource scaler will scale the resource back up and then route the traffic back to the service (by changing the svc 
+selector).
+
+## Getting Started
+This infrastructure designed to be generic and extendable, it can scale any resource.
+All you have to do is implement the resource scaler, the interface between it and this infrastructure's components is 
```

Comment:
```suggestion
All you have to do is implement the specific resource-scaler for your resource. The interface between your resource-scaler and the scale-to-zero infrastructure's components is 
```

Reply:
If you're asking the "user" to implement component X, then introduce component X like you introduced the scaler explaining what is it's function in the flow, also to avoid ambiguities maybe it's best to not refer to it as `resource scaler` but `resource-scaler` or `resourcescaler` everywhere so it will be read as a single term (the word "scale" appear a lot here)